test/integration/client-navigation/test/index.test.js

```diff
@@ -1381,6 +1381,23 @@ describe('Client Navigation', () => {
     expect(value).toBe(false)
   })
 
+  it('should render react components in head client-side', async () => {
```

Comment:
My main concern here is that client-navigation will not work. Could you add a test for that as well? Potentially we'll just have to land this even if it does not work for backwards compatibility.

Reply:
Ah yeh good point, forgot you mentioned that earlier. Working on that now 👍  Frankly, I'm shocked component composition was never supported within next/head; that's a fundamental concept around React dev. Don't get me wrong, I am overall extremely pleased with NextJS and am honored to help work on this OSS, just confused why this was never supported (maybe for good reason).

If anything, I am glad this became an issue because you're saying that it was only ever accidentally supported and probably failed during client-side navigation (yikes). This _may_ have been why I've seen Rollbar errors on said injected head scripts not working (potentially during client-side navigation)?

Reply:
> just confused why this was never supported (maybe for good reason).

We can't use React to render into the head as you'd have to nest the rendering in an element and `<head>` needs tags to be inserted at the top level of the tag. Hence why it's using a custom renderer that does not support custom react components. 

It's documented as part of the `next/head` documentation:

> title, meta or any other elements (e.g. script) need to be contained as direct children of the Head element, or wrapped into maximum one level of <React.Fragment> or arrays—otherwise the tags won't be correctly picked up on client-side navigations.

https://nextjs.org/docs/api-reference/next/head

Reply:
@timneutkens you were right, it fails 😬 I'll need some advice on how to fix that... really confused why we serialize header content in `window.__NEXT_DATA__` and not sure how to proceed.

Reply:
welp, gave it my best shot.